cate/core/opimpl.py

```diff
@@ -71,6 +72,57 @@ def normalize_impl(ds: xr.Dataset) -> xr.Dataset:
     return ds
 
 
+def _normalize_zonal_lat_lon(ds: xr.Dataset) -> xr.Dataset:
+    """
+    In case that the dataset only contains lat_centers and is a zonal mean dataset,
+    the longitude dimension created and filled with the variable value of certain latitude.
+    :param ds: some xarray dataset
+    :return: a normalized xarray dataset
+    """
+
+    if 'latitude_centers' not in ds.coords:
+        return ds
+    if 'lon' in ds.coords:
+        return ds
+    ds_zonal = ds.copy()
+    resolution = (ds.latitude_centers.values[1] - ds.latitude_centers.values[0])
+    ds_zonal = ds_zonal.assign_coords(lon=[i + (resolution / 2) for i in np.arange(-180.0, 180.0, resolution)])
```

Comment:
I very much dislike this part. We only have the one value, so, it creates a large overhead to copy this ~36 times or so. As currently cate cannot deal with cells with different resolutions, we have no choice, though.